src_c/scrap.c

```diff
@@ -355,6 +408,77 @@ _scrap_set_mode(PyObject *self, PyObject *args)
 
 #endif /* !defined(MAC_SCRAP) */
 
+/**
+ * @brief Fetches a python string from the SDL clipboard. If
+ *        there is nothing in the clipboard, it will return empty
+ *
+ * @return PyObject*
+ */
+static PyObject *
+_scrap_get_text(PyObject *self, PyObject *args)
+{
+    char *text = SDL_GetClipboardText();
+
+    // if SDL_GetClipboardText fails, it returns an empty string
+    if (text[0] == '\0') {
+        const char *err = SDL_GetError();
+        if (err) {
+            SDL_free(text);
+            PyErr_SetString(pgExc_SDLError, err);
+            return NULL;
+        }
+    }
+
+    PyObject *returnValue = PyUnicode_FromString(text);
+    SDL_free(text);
+
+    return returnValue;
+}
+
+/**
+ * @brief Puts a python string into the SDL clipboard
+ *
+ * @param args A python string to be put into the clipboard
+ *
+ * @return PyObject*
+ */
+static PyObject *
+_scrap_put_text(PyObject *self, PyObject *args)
+{
+    char *text;
+
+    if (!PyArg_ParseTuple(args, "s", &text)) {
+        return NULL;
+    }
+
+    if (SDL_SetClipboardText(text)) {
+        const char *err = SDL_GetError();
+        if (err) {
+            return RAISE(pgExc_SDLError, err);
```

Comment:
These three lines just need to be `return RAISE(pgExc_SDLError, SDL_GetError());`